tests/playwright/sanity/assets/dev/js/editor/elements/views/widget.test.js

```diff
@@ -0,0 +1,16 @@
+const { test, expect } = require( '@playwright/test' );
+const WpAdminPage = require( '../../../../../../../pages/wp-admin-page' );
+
+test( 'Check if the empty placeholder is displayed in the Image Carousel', async ( { page }, testInfo ) => {
+	// Arrange.
+	const wpAdmin = new WpAdminPage( page, testInfo ),
+		editor = await wpAdmin.useElementorCleanPost();
+
+	// Act.
+	await editor.addWidget( 'image-carousel' );
+
+    const emptyViewPlaceholder = editor.getPreviewFrame().locator( '.elementor-widget-image-carousel .elementor-widget-empty-icon' );
+
+	// Assert.
+    await expect( emptyViewPlaceholder ).toHaveCount( 1 );
```

Comment:
Why use `toHaveCount()`? You could use `toBeVisible()` also.

Reply:
@udidol 

My logic: 
- The `widget.js` file checks if the height of a widget = 0 and if the widget is `:visible`. 
- If so, then the JS file will insert a placeholder. 
- The empty insert placeholder should get a placeholder, as it has the height = 0 and is visible.
- All that I do in my test is to check if the placeholder has been inserted in the empty Image Carousel widget.

In other words, I am only testing the outcome of the `onRender()` method inside `widget.js` code and not the conditions. I am not sure whether that is good practice or not.

Reply:
Ok I get that